core/src/main/java/org/infinispan/commands/tx/PrepareCommand.java

```diff
@@ -230,6 +231,9 @@ public boolean isReturnValueExpected() {
       }
       Set<Object> set = new HashSet<>(modifications.length);
       for (WriteCommand wc : modifications) {
+         if (wc.hasFlag(Flag.SKIP_LOCKING)) {
```

Comment:
I am sorry to piggyback this change on yours.  But it seems it would be more efficient to use a TreeSet if sort is true instead of sorting at the very end using the array.